src/store/modules/user/actions.ts

```diff
@@ -346,27 +346,36 @@ const actions: ActionTree<UserState, RootState> = {
       "viewSize": payload.viewSize
     }
 
-    let users = JSON.parse(JSON.stringify(state.users.list)), total = 0;
-
+    let users = JSON.parse(JSON.stringify(state.users.list)); // Initialize users with the current state of users
+    let total = state.users.total;
+    
     try {
-      const resp = await UserService.fetchUsers(params);
-
-      if (!hasError(resp) && resp.data.count) {
-        if (payload.viewIndex && payload.viewIndex > 0) {
-          users = users.concat(resp.data.docs);
-        } else {
-          users = resp.data.docs;
+      const resp = await UserService.fetchUsers(params); // Fetch users from the service
+      if (!hasError(resp)) { 
+        if (resp.data.count > 0) { 
+          if (payload.viewIndex && payload.viewIndex > 0) { 
+            users = users.concat(resp.data.docs); 
+          } else {
+            users = resp.data.docs; // Replace the users list with the new users
+          }
+          total = resp.data.count; // Update the total number of users
         }
-        total = resp.data.count;
       } else {
-        throw resp.data;
+        throw resp.data; 
+      }
+    } catch (error) {
+      if (payload.viewIndex === 0) { 
+        users = []; 
+        total = 0; 
+      } else {
+        users = state.users.list;
```

Comment:
I think we don't need this check for updating the users when having a viewIndex other than 0.